aws-datastore/src/main/java/com/amplifyframework/datastore/appsync/AppSyncClient.java

```diff
@@ -187,26 +187,26 @@ public static AppSyncClient via(@NonNull GraphQLBehavior api) {
     @NonNull
     @Override
     public <T extends Model> Cancelable delete(
```

Comment:
these are public API changes. will this change affect any customers? Should we follow proper deprecation steps?

Reply:
Also, is it ever necessary to pass in both model instance _and_ model schema? I believe the original method signature was set up this way because we were more reserved about allowing model instances in the parameter early on in the development. In my understanding, that is no longer necessary because we now support `SerializedModel`. So we won't need `ModelSchema` anymore because we can directly pass in the model instance right?

Reply:
> these are public API changes. will this change affect any customers? Should we follow proper deprecation steps?

I guess technically they are `public`, but this class is definitely not intended to be used by customers.  This is just the interface that DataStore uses to interact with API, so I don't think we need to be concerned.

> Also, is it ever necessary to pass in both model instance and model schema?

Hmm, there's a lot to that question, but I think we do still need both.  I'm not entirely sure why, because we should be able to derive `ModelSchema` from a generated `Model`, or a `SerializedModel`.  I vaguely remember suggesting this same thing a while back, but there was some reason I'm forgetting.  Ultimately it seems like we should strive for that.   But for now, this pattern of passing both schema and model extends throughout DataStore, much more than the scope of this PR.